core/src/credential.rs

```diff
@@ -80,4 +95,62 @@ impl Credential {
     pub fn serialize(&self) -> Value {
         return serde_json::to_value(&self).unwrap();
     }
+    
+    pub fn deserialize (contents: String) -> Credential {
```

Comment:
We should probably create a ticket to have these return results `Result<Credential, Error>`. JSON is one of the most error prone parts of the code and it's quite a hot path for panicking.

Reply:
I think, it can be done as a part of this ticket. In fact it is  already done.